crates/zksync/core/src/vm/db.rs

```diff
@@ -28,6 +28,22 @@ use super::storage_recorder::{AccountAccess, AccountAccesses, CallType, StorageA
 /// Default chain id
 pub(crate) const DEFAULT_CHAIN_ID: u32 = 31337;
 
+// NOTE: we use vec instead of hashmap as 2 contracts share the same bytecode hash
+// (BOOTLOADER and 0x00 share "empty contract")
```

Comment:
Does this mean we will return Bootloader bytecode or the empty bytecode? Perhaps we should clean this up, as this is a weird scenario. Else we may eventually wish to decouple our empty bytecode contract from Bootloader by making it ... well not empty `const EMPTY_CONTRACT = [1u32; 32]`.

Also, which `0x00` contract is the comment referring to? AFAIK, empty contract is `32` `0`s long

Reply:
The system contracts "empty contract" is actually _not_ 32 0s, it's a fully fledged bytecode which you can find in [anvil-zksync repo](https://github.com/matter-labs/anvil-zksync/blob/main/crates/core/src/deps/contracts/EmptyContract.json).
That bytecode is then [used](https://github.com/matter-labs/anvil-zksync/blob/35e4a6c895155c46c4add7ae9f7facf29f0dd3ae/crates/core/src/deps/system_contracts.rs#L216-L226) to load the bootloader address and the 0 address (0x00 in the comment) returned by `get_deployed_contracts`.
So, the bytecode hash for 2 addresses will match, which causes the first instance of `DeployedContract`(0x00 in this case) to be omitted in the constructed map, so the override key for the first contract is never made.

Reply:
I will clarify the comment in the code